### PR TITLE
Make dshell/cpp_header_gen.d runnable on windows/osx

### PR DESCRIPTION
**Pull request by MoonlightSentinel**

### CODE REVIEW

File: test/dshell/extra-files/cpp_header_gen/library.d [hunk 1]
```diff
@@ -72,7 +72,7 @@ void toggle(ref U u)
 // FIXME: Generates non-existant global
 // enum PI = 3.141; // extern _d_double PI;
 
-__gshared immutable PI = 3.141;
+__gshared const PI = 3.141;
```
Comment (MoonlightSentinel) [inline test/dshell/extra-files/cpp_header_gen/library.d:75]: DMD rejects `immutable` for `extern(C++)` on windows but not on linux ...

**Review by thewilsonator — APPROVED** [review]

File: test/dshell/cpp_header_gen.d [hunk 2]
```diff
@@ -4,22 +4,16 @@ import dshell;
...
-    {
-        writeln("CPP header generation test was skipped on non-linux platform.");
-        return;
-    }
-
-    // FIXME: Should be a default variable
-    Vars.set("CC", "g++");
-
     Vars.set("SOURCE_DIR",  "$EXTRA_FILES/cpp_header_gen");
-    Vars.set("LIB",         "$OUTPUT_BASE/lib$LIBEXT");
```
Comment (MoonlightSentinel) [inline test/dshell/cpp_header_gen.d:17]: This change isn't strictly necessary but seeing `lib.lib` could be pretty confusing for other readers.

File: test/dshell/cpp_header_gen.d [hunk 3]
```diff
@@ -4,22 +4,16 @@ import dshell;
...
-    Vars.set("CPP_OBJ",     "$OUTPUT_BASE/cpp.o");
+    Vars.set("LIB",         "$OUTPUT_BASE/library$LIBEXT");
```
Comment (MoonlightSentinel) [inline test/dshell/cpp_header_gen.d:8]: This change isn't strictly necessary but seeing `lib.lib` could be pretty confusing for other readers.

File: test/tools/dshell_prebuilt/dshell_prebuilt.d [hunk 4]
```diff
@@ -69,6 +73,11 @@ void dshellPrebuiltInit(string testDir, string testName)
         mixin(`Vars.set("` ~ var ~ `", requireEnv("` ~ var ~ `"));`);
     }
 
+    static foreach (var; optionalEnvVars)
```
Comment (UplinkCoder) [inline test/tools/dshell_prebuilt/dshell_prebuilt.d:76]: awwwww why that?

Comment (MoonlightSentinel) [reply test/tools/dshell_prebuilt/dshell_prebuilt.d:76]: Copy & Paste error. I'll remove the `static`

**Pull request merged**